### PR TITLE
AUTO-204/AUTO-205/AUTO-206 - Automated tests for db cleanup validations - stable/5.0

### PR DESCRIPTION
**Pull request by nasharatm**
Covers: encrypted, unencrypted volumes/workloads as well. 
workload: default and scheduled 
=========
tempest.api.workloadmgr.workload.test_workload.WorkloadTest.test_3_create_workload 1
tempest.api.workloadmgr.workload.test_workload.WorkloadTest.test_12_create_scheduled_workload 1

snapshots: full, incr
=========
tempest.api.workloadmgr.snapshot.test_bootfromvol_fullsnapshot 1
tempest.api.workloadmgr.snapshot.test_snapshots 2

restore: selective, in-place, oneclick
=========
tempest.api.workloadmgr.restore.test_tvault1040_oneclick_restore 1
tempest.api.workloadmgr.barbican.test_barbican.WorkloadsTest.test_17_barbican 3

[db-cleanup-wl-snapshots-restore.pdf](https://github.com/trilioData/tempest/files/9392318/db-cleanup-wl-snapshots-restore.pdf)
[barbican-dbcleanup-all-restores-results.pdf](https://github.com/trilioData/tempest/files/9392319/barbican-dbcleanup-all-restores-results.pdf)

### CODE REVIEW

File: tempest/api/workloadmgr/base.py [hunk 1]
```diff
@@ -4240,3 +4240,58 @@ def remove_role_from_user_project(self, project_id, user_id, role_id):
...
 
 
+    '''
+    Method to get db data for workload validations - DB cleanup
+    '''
+    def db_cleanup_workload_validations(self):
+        workload_validations = {}
+        LOG.debug("Getting list for db cleanup workload validations.")
+
+        workload_tables = ["workload_vms", "workload_vm_metadata", "scheduled_jobs", "snapshots"]
```
Comment (savitha-p) [inline tempest/api/workloadmgr/base.py]: Move this list to tvaultconf.py

File: tempest/api/workloadmgr/base.py [hunk 2]
```diff
@@ -4240,3 +4240,58 @@ def remove_role_from_user_project(self, project_id, user_id, role_id):
...
+        return workload_validations
+
+    '''
+    Method to get db data for snapshot validations - DB cleanup
+    '''
+    def db_cleanup_snapshot_validations(self):
+        snapshot_validations = {}
+        LOG.debug("Getting list for db cleanup snapshot validations.")
+
+        snapshot_tables = ["snapshots", "snapshot_metadata", "vm_recent_snapshot", "snapshot_vm_resources",
```
Comment (savitha-p) [inline tempest/api/workloadmgr/base.py]: Move this tables list to tvaultconf.py

File: tempest/api/workloadmgr/base.py [hunk 3]
```diff
@@ -4240,3 +4240,58 @@ def remove_role_from_user_project(self, project_id, user_id, role_id):
...
+        return snapshot_validations
+
+    '''
+    Method to get db data for restore validations - DB cleanup
+    '''
+    def db_cleanup_restore_validations(self):
+        restore_validations = {}
+        LOG.debug("Getting list for db cleanup restore validations.")
+
+        restore_tables = ["restore_metadata", "restored_vm_metadata", "restored_vm_resource_metadata", "restored_vm_resources", "restored_vms", "restores"]
```
Comment (savitha-p) [inline tempest/api/workloadmgr/base.py]: Move this tables list to tvaultconf.py

File: tempest/api/workloadmgr/workload/test_workload.py [hunk 4]
```diff
@@ -132,10 +135,21 @@ def test_3_create_workload(self):
             else:
                 reporting.add_test_step("Create workload", tvaultconf.FAIL)
                 reporting.set_test_script_status(tvaultconf.FAIL)
-
+            
+            self.db_cleanup_workload_validations()
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload.py]: Is this call required ?

File: tempest/api/workloadmgr/workload/test_workload.py [hunk 5]
```diff
@@ -652,12 +669,24 @@ def test_12_create_scheduled_workload(self):
...
-                reporting.add_test_step("Verification", tvaultconf.PASS)
+                reporting.add_test_step("Verification for workload schedule", tvaultconf.PASS)
                 LOG.debug("Workload schedule enabled")
             else:
-                reporting.add_test_step("Verification", tvaultconf.FAIL)
+                reporting.add_test_step("Verification for workload schedule", tvaultconf.FAIL)
                 LOG.error("Workload schedule not enabled")
+        
+            # Delete workload
+            self.workload_delete(self.wid)
```
Comment (savitha-p) [inline tempest/api/workloadmgr/workload/test_workload.py:680]: When you are explicitly deleting the workload, you need to pass workload_cleanup=False in workload_create() call.

File: tempest/api/workloadmgr/workload/test_workload.py (hunk 4 above)
Comment (nasharatm) [reply tempest/api/workloadmgr/workload/test_workload.py]: removed.

File: tempest/api/workloadmgr/workload/test_workload.py (hunk 5 above)
Comment (nasharatm) [reply tempest/api/workloadmgr/workload/test_workload.py:680]: Added change

File: tempest/api/workloadmgr/base.py (hunk 3 above)
Comment (nasharatm) [reply tempest/api/workloadmgr/base.py]: done

File: tempest/api/workloadmgr/base.py (hunk 2 above)
Comment (nasharatm) [reply tempest/api/workloadmgr/base.py]: done

File: tempest/api/workloadmgr/base.py (hunk 1 above)
Comment (nasharatm) [reply tempest/api/workloadmgr/base.py]: done

**Comment by nasharatm** [comment]
Added changes as per review comment and modified queries.
PFA the run results with latest code changes.

[1-dbcleanup-barbican-results.pdf](https://github.com/trilioData/tempest/files/9413663/1-dbcleanup-barbican-results.pdf)
[2-dbcleanup-snapshot-wl-restore-results.pdf](https://github.com/trilioData/tempest/files/9413664/2-dbcleanup-snapshot-wl-restore-results.pdf)

**Review by priyankagupte15 — APPROVED** [review]

File: tempest/api/workloadmgr/snapshot/test_snapshots.py [hunk 6]
```diff
@@ -49,6 +49,7 @@ def test_1_create_full_snapshot(self):
             global volume_id
             global workload_id
             global snapshot_id
+            global full_snapshot_validations_before
```
Comment (savitha-p) [inline tempest/api/workloadmgr/snapshot/test_snapshots.py]: where is this global variable getting used ?

**Review by savitha-p — APPROVED** [review]

File: tempest/api/workloadmgr/snapshot/test_snapshots.py (hunk 6 above)
Comment (nasharatm) [reply tempest/api/workloadmgr/snapshot/test_snapshots.py]: It was used previously, removed as no longer needed.

**Review by savitha-p — APPROVED** [review]

**Pull request merged**